### ISSUE TITLE
appimage for arm?

### ISSUE THREAD

**Issue by Thatoo** [nanu-c/axolotl#321]
Appimage is now available for x86 in this repo and very soon on appimagehub : https://github.com/AppImage/appimage.github.io/pull/2431

Axolotl is available for arm as a click package.

Would it be possible to get an arm appimage package also for system like mobian, arm arch etc... ?
I guess the most difficult has already been done.

In this way, axolotl will be the only signal app available for arm on all linux distribution (Mobian, POstmarketOS, arch, Ubuntu Touch...)

Regards,

**Comment by Thatoo** [comment]
Actually, with both Appimage, Axolotl will be the only signal app to be available on all linux installation basically which is not the case for the original signal app.

**Comment by nanu-c** [comment]
@olof-nord  can you add a arm64 github pipeline for the appimages?

**Comment by Thatoo** [comment]
That would be awesome!

I just realised that if v0.9.4 had an Axolotl-0.9.4-x86_64.AppImage.zip package, alongside 3 other .zip package (macos.zip, ubuntu.zip, web.zip, AppImage.zip), v0.9.6 and v0.9.8 have only two .click package...
Is it not supposed to be automatically creating the 4 .zip package at the same time than the 2 .click package for each release?

**Comment by olof-nord** [comment]
To build for arm, afaik, we would need an ARM-powered GitHub worker for the CI pipeline.
We are currently building for x86_64, as that is the platform provided.

In the below Signal-Desktop Issue thread, @rathboma offers just that for the Signal-Desktop builds.

> Maybe I could host a GitHub worker on that box for you? The box has plenty of spare capacity. That way you can support ARM builds and I can get Signal on my Pi4 :-D.

https://github.com/signalapp/Signal-Desktop/issues/3410#issuecomment-692724499

I just used the contact form on @rathboma's own website and asked him about a possible collaboration regarding this.
Will keep this issue as well as the Axolotl Telegram Group posted on progress.

**Comment by rathboma** [comment]
Hey! Yeah I have an arm64 box I could host a worker on if you like :-) it's not super powered, but gets the job done.

**Comment by olof-nord** [comment]
> Hey! Yeah I have an arm64 box I could host a worker on if you like :-) it's not super powered, but gets the job done.

Exiting! We have had five releases so far this year, I dont expect any large usage, just to allow us to automate the release process.

From reading about how to set this up, I believe @rathboma needs the token from this repo, and we need the name of your runner. Here a link to the documentation: https://docs.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners

I do unfortunately not have access to the settings page of this repository, here @nanu-c needs to help us out.

@nanu-c, could you please go here https://github.com/nanu-c/axolotl/settings/actions/add-new-runner and let @rathboma know the command (including the token) listed under Configure? Not sure the best way to communicate here, but I believe a bit more private than per public github issue comments would be the best option.

**Comment by nanu-c** [comment]
@rathboma your offer is very welcome, how can I contact you?

**Comment by Flaburgan** [comment]
> I just realised that if v0.9.4 had an Axolotl-0.9.4-x86_64.AppImage.zip package, alongside 3 other .zip package (macos.zip, ubuntu.zip, web.zip, AppImage.zip), v0.9.6 and v0.9.8 have only two .click package...
> Is it not supposed to be automatically creating the 4 .zip package at the same time than the 2 .click package for each release?

No, I did this release and built locally all the binaries before adding them to the github release. This process isn't automated (yet) but thanks to @olof-nord we could have this, and it would be very great.

**Comment by rathboma** [comment]
Send me an email - firstname dot lastname gmail :-)

**Comment by rathboma** [comment]
Oh, also if you want your own in the medium term, AWS has an 'open source credits' program - https://aws.amazon.com/blogs/opensource/aws-promotional-credits-open-source-projects/

**Comment by Thatoo** [comment]
appimage is back for x86 on last version 0.9.9. Awesome! Only the arm appimage is missing.

**Comment by olof-nord** [comment]
> appimage is back for x86 on last version 0.9.9. Awesome! Only the arm appimage is missing.

Happy to read! Yes, from now on the release process is more automated, and a x86 AppImage is added.

Still working on a more structured AppImage build process, trying to make use of the AppImageBuilder.yml "manifest" but so far without luck. The idea being that this recipe could be executed on different architectures in an easy way.

https://github.com/nanu-c/axolotl/compare/main...olof-nord:add-appimage-builder
